pkg/scheduler/core/division_algorithm.go

```diff
@@ -48,7 +51,7 @@ func divideReplicasByResource(
 	} else if assignedReplicas < spec.Replicas {
 		// We need to enlarge the replicas in terms of the previous result (if exists).
 		// First scheduling is considered as a special kind of scaling up.
-		newTargetClusters, err := scaleUpScheduleByReplicaDivisionPreference(clusters, spec, preference)
+		newTargetClusters, err := scaleUpScheduleByReplicaDivisionPreference(clusters, spec, preference, scheduledClusters, assignedReplicas)
 		if err != nil {
 			return nil, fmt.Errorf("failed to scaleUp: %v", err)
 		}
```

Comment:
what if assignedReplicas == spec.Replicas, but len(scheduledClusters) < len(spec.clusters)

Reply:
I didn't think of a similar rescheduling scenario

Reply:
> what if assignedReplicas == spec.Replicas, but len(scheduledClusters) < len(spec.clusters)

I didn't think of a similar rescheduling scenario

Reply:
Could you please give an example? It seems not possible to happen "in one event"

Reply:
Could you please give an example? It seems not possible to happen "in one event"

Reply:
Let's say we have 2 clusters [A、B] , both with allocatable resources: 4C8G
a deployment requests: 2C4G, 3 replicas
and the propagationPolicy is:
```
placement:
    clusterAffinity:
      clusterNames:
      - A
      - B
    replicaScheduling:
      replicaDivisionPreference: Weighted
      replicaSchedulingType: Divided
      weightPreference:
        dynamicWeight: AvailableReplic
```
so the deployment was placed in both A and B, let's assume A has 2 replicas and B has 1。

1. update propagationPolicy placement.clusterAffinity.clusterNames=[A], now no enough resources for this deployment to reschedule, so nothing happened;
2. update deployment.spec.replicas=2, with this pr, nothing happened again..

Reply:
you are right that it should return the scheduledClusters as the result